spec/network_interface/network_interface_generic_spec.rb

```diff
@@ -0,0 +1,114 @@
+describe LinuxAdmin::NetworkInterfaceGeneric do
```

Comment:
This should really be testing the base class, right?

You could instantiate one of the child classes to test the instance methods.